### ISSUE TITLE
Converting MPT checkpoint to HF format breaks 

### ISSUE THREAD

**Issue by eldarkurtic** [mosaicml/llm-foundry#554]
When trying to convert MPT checkpoint from Composer to HF with `convert_composer_to_hf.py`, it breaks with:
```shell
Traceback (most recent call last):
  File "/mypath/llm-foundry/scripts/inference/convert_composer_to_hf.py", line 344, in <module>
    convert_composer_to_hf(parse_args())
  File "/mypath/llm-foundry/scripts/inference/convert_composer_to_hf.py", line 232, in convert_composer_to_hf
    AutoConfig.register('mpt', MPTConfig)
  File "/mypath/miniconda3/envs/sml/lib/python3.10/site-packages/transformers/models/auto/configuration_auto.py", line 1049, in register
    CONFIG_MAPPING.register(model_type, config)
  File "/nfs/scistore14/alistgrp/ekurtic/miniconda3/envs/sml/lib/python3.10/site-packages/transformers/models/auto/configuration_auto.py", line 753, in register
    raise ValueError(f"'{key}' is already used by a Transformers config, pick another name.")
ValueError: 'mpt' is already used by a Transformers config, pick another name.
```

**Comment by dakinggg** [comment]
To unblock you, please do not use the latest version of `transformers`. That is the version in which they added an MPT implementation to `transformers`.

**Comment by dakinggg** [comment]
This is resolved by #561. We only support the version of MPT in this repo, which requires using `trust_remote_code: True`.

**Comment by eldarkurtic** [comment]
Sorry for the late reply, thanks a lot @dakinggg !
